acceptance-tests/src/test-support/java/tech/pegasys/artemis/test/acceptance/dsl/ArtemisNode.java

```diff
@@ -137,39 +138,70 @@ public UnsignedLong getGenesisTime() throws IOException {
     return fetchGenesisTime();
   }
 
-  public void waitForNewBlock() throws IOException {
-    final Bytes32 startingBlockRoot = getCurrentBeaconHead().getBlockRoot();
-    waitFor(
-        () -> assertThat(getCurrentBeaconHead().getBlockRoot()).isNotEqualTo(startingBlockRoot));
+  public void waitForNewBlock() {
+    final Bytes32 startingBlockRoot = waitForBeaconHead().getBlockRoot();
+    waitFor(() -> assertThat(waitForBeaconHead().getBlockRoot()).isNotEqualTo(startingBlockRoot));
   }
 
-  public void waitForNewFinalization() throws IOException {
-    UnsignedLong startingFinalizedEpoch = getChainHead().finalized_epoch;
+  public void waitForNewFinalization() {
+    UnsignedLong startingFinalizedEpoch = waitForChainHead().finalized_epoch;
     LOG.debug("Wait for finalized block");
     waitFor(
-        () -> assertThat(getChainHead().finalized_epoch).isNotEqualTo(startingFinalizedEpoch), 540);
+        () -> assertThat(waitForChainHead().finalized_epoch).isNotEqualTo(startingFinalizedEpoch),
```

Comment:
Probably should use `fetchChainHead` here inside the waiter.
```suggestion
        () -> assertThat(fetchChainHead().finalized_epoch).isNotEqualTo(startingFinalizedEpoch),
```